src/main/java/org/opensha/sha/earthquake/faultSysSolution/ruptures/util/SectIDRange.java

```diff
@@ -10,53 +13,70 @@
  *
  */
 public abstract class SectIDRange implements Comparable<SectIDRange> {
-	
+
+	private final static Map<SectIDRange, SectIDRange> cache = new HashMap<>();
```

Comment:
Should this be a ConcurrentHashMap since build() isn't synchronized?

Reply:
Good call. I've changed it. Doesn't seem to impact performance.